.circleci/config.yml

```diff
@@ -74,8 +92,10 @@ jobs:
 workflows:
   commit:
     jobs:
+      - common-precommit-checks
       - controller-test
       - gui-static-analysis
       - gui-build-image:
           requires:
+            - common-precommit-checks
             - gui-static-analysis
```

Comment:
Was talking to Hugh about this over Slack; lgtm.

Only one comment from the peanut gallery:

```
      - gui-build-image:
          requires:	          requires:
            - common-precommit-checks
            - gui-static-analysis	            - gui-static-analysis
```

might not want to add the pre-commit checks to this `requires`, and in fact I'd recommend getting rid of the `requires` altogether.  This has the effect of adding latency to the build by serializing it!

Reply:
FWIW it's possible to cache the pip store and the pre-commit store.  Dunno how much it matters, though, and it can always be done later.  Getting everything onto CircleCI will be a big improvement in itself.

Reply:
that's true, but this job takes 13 seconds on CI. I'd rather we failed fast instead of of clogging up our CI pipeline with docker builds (that take minutes)